examples/1_basics/7_parallelization_cluster.py

```diff
@@ -0,0 +1,112 @@
+SLURM cluster. If you do not want to use a cluster but your local machine, set dask_client
+to `None` and pass `n_workers` to the `Scenario`.
+
+We use the black-box facade because it is designed for black-box function optimization.
+The black-box facade uses a :term:`Gaussian Process<GP>` as its surrogate model.
+The facade works best on a numerical hyperparameter configuration space and should not
+be applied to problems with large evaluation budgets (up to 1000 evaluations).
+"""
+
+import numpy as np
+from ConfigSpace import Configuration, ConfigurationSpace, Float
+from dask.distributed import Client
+from dask_jobqueue import SLURMCluster
+from matplotlib import pyplot as plt
+
+from smac import BlackBoxFacade, RunHistory, Scenario
+
+__copyright__ = "Copyright 2022, AutoML.org Freiburg-Hannover"
+__license__ = "3-clause BSD"
+
+
+class Branin(object):
+    @property
+    def configspace(self) -> ConfigurationSpace:
+        cs = ConfigurationSpace(seed=0)
+        x0 = Float("x0", (-5, 10), default=-5, log=False)
+        x1 = Float("x1", (0, 15), default=2, log=False)
+        cs.add_hyperparameters([x0, x1])
+
+        return cs
+
+    def train(self, config: Configuration, seed: int = 0) -> float:
+        """Branin function
+
+        Parameters
+        ----------
+        config : Configuration
+            Contains two continuous hyperparameters, x0 and x1
+        seed : int, optional
+            Not used, by default 0
+
+        Returns
+        -------
+        float
+            Branin function value
+        """
+        x0 = config["x0"]
+        x1 = config["x1"]
+        a = 1.0
+        b = 5.1 / (4.0 * np.pi**2)
+        c = 5.0 / np.pi
+        r = 6.0
+        s = 10.0
+        t = 1.0 / (8.0 * np.pi)
+        ret = a * (x1 - b * x0**2 + c * x0 - r) ** 2 + s * (1 - t) * np.cos(x0) + s
+
+        return ret
+
+
+if __name__ == "__main__":
+    model = Branin()
+
+    # Scenario object specifying the optimization "environment"
+    scenario = Scenario(model.configspace, deterministic=True, n_trials=100)
+
+    # Create cluster
+    n_workers = 4  # Use 4 workers on the cluster
+
+    cluster = SLURMCluster(
+        # This is the partition of our slurm cluster.
+        queue="cpu_short",
+        # Your account name
+        # account="myaccount",
+        cores=1,
+        memory="1 GB",
+        # Walltime limit for each worker. Ensure that your function evaluations
+        # do not exceed this limit.
+        # More tips on this here: https://jobqueue.dask.org/en/latest/advanced-tips-and-tricks.html#how-to-handle-job-queueing-system-walltime-killing-workers
+        walltime="00:30:00",
+        processes=1,
+        log_directory="tmp/smac_dask_slurm",
+    )
+    cluster.scale(jobs=n_workers)
+
+    # Dask will create n_workers jobs on the cluster which stay open.
+    # Then, SMAC/Dask will schedule individual runs on the workers like on your local machine.
+    client = Client(
+        address=cluster,
+    )
```

Comment:
You can also do `cluster.get_client()` which is a bit easier and requires one less import :)

Reply:
I wasn't aware of that, added a note :)